Find-Package.NuGet.ArgumentCompleters.ps1

```diff
@@ -0,0 +1,84 @@
+	Installation:
+	* Using PowerShell v5+:
+		Invoke Find-Package.NuGet.ArgumentCompleters.ps1 (or add it to your profile)
+	* TabExpansionPlusPlus:
+		Find-Package.NuGet.ArgumentCompleters.ps1 should be placed in the same directory as TabExpansionPlusPlus. Optionally, invoke Find-Package.NuGet.ArgumentCompleters.ps1 (or add it to your profile)
+	* TabExpansion2.ps1
+		Find-Package.NuGet.ArgumentCompleters.ps1 should be added to the path. Optionally, AFTER invoking TabExpansion2.ps1, invoke Find-Package.NuGet.ArgumentCompleters.ps1 (or add it to your profile)
+#>
+Register-ArgumentCompleter -CommandName Find-Package -ParameterName Name -ScriptBlock {
+	param($commandName, $parameterName, $wordToComplete, $commandAst, $boundParameters)
+
+	# Check that the loaded PackageManagement module is new enough
+	if (-not ("Microsoft.PackageManagement.NuGetProvider.RequestWrapper" -as [Type])) {
+		return
+	}
+
+	# This is a workaround until we can figure out how to properly page results. Currently argument completers seem to fully enumerate the results, so we have to limit the number of results ourselves.
+	# Change this if you want more results, but be aware that the script's signing becomes invalid.
+	$MAX_COMPLETION_RESULTS = 20
+
+
+    $location = $null # String
+    $credentialUserName = $null # String
+    $credentialUserPassword = $null # SecureString
+    $proxy = $null # IWebProxy
+    $headers = $null # string[]
+
+    if ($boundParameters.ContainsKey('Source')) {
+        $sourceValue = $boundParameters['Source']
+        $ps = Get-PackageSource $sourceValue -ErrorAction Ignore -WarningAction Ignore
+        if ($ps -ne $null) {
+            $location = $ps.Location
+        } elseif ([System.Uri]::IsWellFormedUriString($sourceValue, [System.UriKind]::Absolute)) {
+            $location = $sourceValue
+        }
+    } else {
+        $ps = Get-PackageSource -ErrorAction Ignore -WarningAction Ignore
+        if ($ps -and ($ps.Count -eq 1)) {
+            $location = $ps.Location
+        }
+    }
+
+    if ($boundParameters.ContainsKey('Credential')) {
+        $credentialUserName = $boundParameters['Credential'].UserName
+        $credentialUserPassword = $boundParameters['Credential'].Password
+    }
+
+    if ($boundParameters.ContainsKey('Headers')) {
+        $headers = $boundParameters['Headers']
+    }
+
+    # TODO: Handle proxy credentials for full and core CLR
+    if ($boundParameters.ContainsKey('Proxy')) {
+        $proxyUrl = $boundParameters['Proxy']
+        $proxy = [System.Net.WebProxy]::new($proxyUrl)
+    }
+
+    if ($location)
+    {
+        $fakeRequest = New-Object -TypeName Microsoft.PackageManagement.NuGetProvider.RequestWrapper
+        $resources = [Microsoft.PackageManagement.NuGetProvider.NuGetResourceCollectionFactory]::GetResources($location, $fakeRequest)
```

Comment:
>[Microsoft.PackageManagement.NuGetProvider.NuGetResourceCollectionFactory]::GetResources($location, $fakeRequest) [](start = 21, length = 113)

Is this api handles v2, v3 and local directory locations?

Reply:
Yes - Check the NuGetResourceCollectionFactory class
v2 and local directory should not discover an AutoComplete feed

---
In reply to: [154492336](https://github.com/OneGet/NuGetProvider/pull/25#discussion_r154492336) [](ancestors = 154492336)